core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java

```diff
@@ -736,6 +791,12 @@ public void close() {
     if (tableZooHelper != null) {
       tableZooHelper.close();
     }
+    if (scannerReadaheadPool != null) {
```

Comment:
Either need to make this variable volatile or synchronize this method to ensure changes to the variable made by different threads are seen.

Reply:
@keith-turner  Do you believe this is an existing problem with the thriftTransportPool also?

Reply:
I synchronized the `close` method in 47e90dca559f02683c7c9f58ae366a422367a2d7

Reply:
I'm not sure it's necessary, since the `closed` variable is volatile again. As long as you add `ensureOpen();` to the two methods that retrieve the pools. That said, it's not like we're calling `close()` so frequently that synchronizing it is going to matter.

Reply:
Yeah closed does not need to be efficient, no need to worry about thread contention here.